kasa/smartdevice.py

```diff
@@ -289,20 +289,32 @@ async def get_sys_info(self) -> Dict[str, Any]:
         """Retrieve system information."""
         return await self._query_helper("system", "get_sysinfo")
 
+    def _update_query(self, include_emeter=True) -> Dict[str, Dict[str, Any]]:
+        req: Dict[str, Dict[str, Any]] = {}
+        if include_emeter:
+            req.update(self._create_emeter_request())
+        req.update(self._create_request("system", "get_sysinfo"))
+        return req
+
+    async def _update_with_request(self, request):
+        self._last_update = await self.protocol.query(self.host, request)
+        self._sys_info = self._last_update["system"]["get_sysinfo"]
+
     async def update(self):
         """Update some of the attributes.
 
         Needed for methods that are decorated with `requires_update`.
         """
-        req = {}
-        req.update(self._create_request("system", "get_sysinfo"))
+        # Some devices without an emeter will crash if their emeter status
+        # is queried. Therefore, ensure the first update query is only for
+        # get_sysinfo.
+        if self._last_update is None:
+            initial_update_req = self._update_query(include_emeter=False)
+            await self._update_with_request(initial_update_req)
```

Comment:
Save the response somewhere temporarily, that way there is no need to do the second query in the case if there's no emeter found.

Having two requests when the emeter is found is fine (as second one should've been made anyway).

Otherwise this looks good to me, but please add a unit test to check the behavior to make sure no future refactoring will break this behavior!